src/3d/qgsrubberband3d.cpp

```diff
@@ -402,9 +415,20 @@ void QgsRubberBand3D::moveLastPoint( const QgsPoint &pt )
 
 void QgsRubberBand3D::updateGeometry()
 {
+  // figure out a reasonable origin for the coordinates to keep them as small as possible
+  QgsBox3D box;
+  QgsVertexIterator vit = mGeometry.vertices();
+  while ( vit.hasNext() )
+  {
+    const QgsPoint pt = vit.next();
+    box.combineWith( pt.x(), pt.y(), pt.z() );
+  }
```

Comment:
This will be much more efficient:

```suggestion
  const QgsBox3D box = mGeometry.constGet()->boundingBox3D();
```